autometa/taxonomy/ncbi.py

```diff
@@ -202,62 +206,56 @@ def get_lineage_dataframe(self, taxids, fillna=True):
                 else:
                     ranked_taxids[taxid].update({rank: name})
         df = pd.DataFrame(ranked_taxids).transpose()
-        df.index.name = 'taxid'
+        df.index.name = "taxid"
         if fillna:
-            df.fillna(value='unclassified', inplace=True)
+            df.fillna(value="unclassified", inplace=True)
         return df
 
     def rank(self, taxid):
         """
-        Takes a taxid and returns its rank
+        Return the respective rank of provided `taxid`. If the `taxid` is deprecated, suppressed,
+        withdrawn from NCBI (basically old) the updated rank will be retrieved
 
         Parameters
         ----------
         taxid : int
-            identifer for a taxon in the Taxonomy Database by NCBI
-            `taxid` to retrieve rank from `nodes` or `merged`
+            `taxid` to retrieve rank from nodes.dmp
 
         Returns
         -------
         str
-            rank name if taxid is found in nodes.dmp or merged.dmp else 'unclassified'
+            rank name if taxid is found in nodes.dmp else "unclassified"
+
+        Raises
+        ------
+        ValueError
+            Provided `taxid` is not a positive integer
         """
-        if not self.is_valid_taxid(taxid):
-            return False
-        # We have already checked that the taxid is present in either 'nodes.dmp' or 'merged.dmp'
-        # in the `is_valid_taxid() function` thus we don't need a default value
-        rank_name = self.nodes.get(taxid).get('rank')
-        if rank_name == None:
-            return self.merged.get(taxid).get('rank')
-        return rank_name
+        taxid = self.is_valid_taxid(taxid)
```

Comment:
Once again, the `is_valid_taxid` method does not check anything to do with `merged.dmp`, so that should be corrected (see other comments). As written now, the behavior of this method does not fit the docstring.